pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,234 @@
+"""
+Signal processing tools for Pynapple.
+
+"""
+
+from itertools import repeat
+
+import numpy as np
+
+import pynapple as nap
+
+
+def compute_fft(sig, fs):
+    """
+
+    Parameters
+    ----------
+    sig : pynapple.Tsd
+        The signal time series to obtain time frequency decomposition from
+    fs : int / float
+        The sampling frequency of sig
+
+    Returns
+    -------
+    fft_result : np.ndarray
+        fourier decomposition signal
+    fft_freq : mp.ndarray
+        fourier frequencies
+    """
+    if not isinstance(sig, nap.Tsd):
+        raise TypeError("Currently compute_fft is only implemented for Tsd")
+    fft_result = np.fft.fft(sig.values)
+    fft_freq = np.fft.fftfreq(len(sig.values), 1 / fs)
+    return fft_result, fft_freq
+
+
+def morlet(M, ncycles=5.0, scaling=1.0):
```

Comment:
I would call it `compute_morlet_wavelet`

Reply:
Also it would be nice in the notebook to show a plot of it.

Reply:
plot will be added in API notebook